[TT-14170] Update documentation for master Gateway (#6061)
TykTechnologies/tyk-docs: tyk-docs/content/shared/gateway-config.md

> ENV: <b>TYK_GW_CONTROLAPIPORT</b><br />

Reviewer: Hey, @lghiur

Similar to dashboard,

Could you check why the streaming configuration isn't available here? I'm specifically looking for this variable, TYK_GW_STREAMING_ENABLED. There's also a [Jira](https://tyktech.atlassian.net/browse/DX-1851) ticket open for the same issue.

Type: `int`<br />

> Set to run your Gateway Control API on a separate port, and protect it behind a firewall if needed. Please make sure you follow this guide when setting the control port https://tyk.io/docs/planning-for-production/#change-your-control-port.

Reviewer 2: ```suggestion
Set this to expose the Tyk Gateway API on a separate port. You can protect it behind a firewall if needed. Please make sure you follow this guide when setting the control port https://tyk.io/docs/planning-for-production/#change-your-control-port.
```

ENV: <b>TYK_GW_CLOUD</b><br />
Type: `bool`<br />

> Cloud flag shows the Gateway runs in Tyk-cloud.

Reviewer 2: ```suggestion
Cloud flag indicates that the Gateway is deployed in a Tyk Cloud data plane.
```
